libraries/AP_Proximity/AP_Proximity_Backend.cpp

```diff
@@ -77,20 +77,41 @@ float AP_Proximity_Backend::correct_angle_for_orientation(float angle_degrees) c
     return wrap_360(angle_degrees * angle_sign + frontend.get_yaw_correction(state.instance));
 }
 
-// check if a reading should be ignored because it falls into an ignore area or if obstacle is near land
-bool AP_Proximity_Backend::ignore_reading(uint16_t angle_deg, float distance_m) const
+// check if a reading should be ignored because it falls into an ignore area (check_for_ign_area should be sent as false if this check is not needed)
+// pitch is the vertical body-frame angle (in degrees) to the obstacle (0=directly ahead, 90 is above the vehicle)
+// yaw is the horizontal body-frame angle (in degrees) to the obstacle (0=directly ahead of the vehicle, 90 is to the right of the vehicle)
+// Also checks if obstacle is near land or out of range
+// angles should be in degrees and in the range of 0 to 360, distance should be in meteres
+bool AP_Proximity_Backend::ignore_reading(float yaw, float pitch, float distance_m, bool check_for_ign_area) const
```

Comment:
let's rename this to "check_reading" or "validate_reading" or something similar.. a positive word instead of a negative.